core/trino-main/src/main/java/io/trino/cost/ComparisonStatsCalculator.java

```diff
@@ -239,4 +250,93 @@ private static PlanNodeStatsEstimate estimateExpressionNotEqualToExpression(
+            SymbolStatsEstimate leftExpressionStatistics,
+            Optional<Symbol> leftExpressionSymbol,
+            SymbolStatsEstimate rightExpressionStatistics,
+            Optional<Symbol> rightExpressionSymbol,
+            boolean isNonEstimatablePredicateApproximationEnabled)
+    {
+        if (leftExpressionStatistics.isUnknown() || rightExpressionStatistics.isUnknown()) {
+            return PlanNodeStatsEstimate.unknown();
+        }
+
+        StatisticRange leftRange = StatisticRange.from(leftExpressionStatistics);
+        StatisticRange rightRange = StatisticRange.from(rightExpressionStatistics);
+        if (leftRange.isEmpty() || rightRange.isEmpty()) {
+            return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+        }
+        // We don't know the correlation between NULLs, so we take the max nullsFraction from the expression statistics
+        // to make a conservative estimate for nullsFilterFactor
+        double nullsFilterFactor;
+        if (isNaN(leftExpressionStatistics.getNullsFraction()) && isNaN(rightExpressionStatistics.getNullsFraction())) {
+            nullsFilterFactor = 1.0;
+        }
+        else {
+            nullsFilterFactor = 1 - maxExcludeNaN(leftExpressionStatistics.getNullsFraction(), rightExpressionStatistics.getNullsFraction());
+        }
+
+        switch (operator) {
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+                // left is always lesser than right
+                if (leftRange.getHigh() < rightRange.getLow()) {
+                    PlanNodeStatsEstimate.Builder estimate = PlanNodeStatsEstimate.buildFrom(inputStatistics);
+                    leftExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            leftExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    rightExpressionSymbol.ifPresent(symbol -> estimate.addSymbolStatistics(
+                            symbol,
+                            rightExpressionStatistics.mapNullsFraction(nullsFraction -> 0.0)));
+                    return estimate.setOutputRowCount(inputStatistics.getOutputRowCount() * nullsFilterFactor)
+                            .build();
+                }
+                // left is always greater than right, no overlap
+                if (leftRange.getLow() > rightRange.getHigh()) {
+                    return inputStatistics.mapOutputRowCount(rowCount -> 0.0);
+                }
+                if (isNonEstimatablePredicateApproximationEnabled) {
+                    double overlappingRangeFraction = leftRange.overlapPercentWith(leftRange.intersect(rightRange));
+                    double alwaysLessRangeFraction = 0.0;
+                    if (leftRange.getLow() < rightRange.getLow()) {
+                        alwaysLessRangeFraction = leftRange.overlapPercentWith(new StatisticRange(leftRange.getLow(), rightRange.getLow(), NaN));
```

Comment:
I don't think intersect works well `NaN` NDVs. 

Why not to do `1.0 - leftRange.overlapPercentWith(rightRange)`?

Reply:
That would include the part of `leftRange` that is greater than `rightRange` (e.g. x = [-10, 10], y = [0, 5]), we want to avoid that.